packages/openneuro-app/src/scripts/datalad/dashboard/datasets/dataset-query.jsx

```diff
@@ -64,6 +65,9 @@ export const datasetQueryDisplay = (isPublic, isSaved) => ({
   variables,
   error,
 }) => {
+  useEffect(() => {
+    if (error) throw error
+  }, [error])
```

Comment:
This is basically the same as setting errorPolicy = 'none' in the Query component. If we can't render the partial result at all, I'd just change that and avoid useEffect throwing errors.

Reply:
When I tried setting errorPolicy to 'none' the errors weren't caught in the appropriate error boundary. For example in the dataset/dataset-query, an undefined dataset would cause error when it's id property was accessed, rather than displaying the "dataset does not exist" error message from the server.

I've moved it to DatasetTab to fix the useEffect hook activation (doesn't always work in an HOC) and updated the error handling to display the results if datasets are returned with the query. Error boundary is activated with server's error message when datasets are undefined.

Reply:
When it's set to none, all query errors will need to be handled by the child of the query component when it receives the error prop. Network errors we could handle with an error link instead, that might be a better approach than using React error boundaries to handle non-React issues? It's not clear to me if a query error is available in the link too, that could be a nice way to catch both cases generally.